app/codes/p2p/sync_chain.py

```diff
@@ -29,10 +34,24 @@ def receive_block(block):
     block_index = block['block_index'] if 'block_index' in block else block['index']
     if block_index > get_last_block_index() + 1:
         sync_chain_from_peers()
+    validate_block(block)
     blockchain.add_block(block)
     return True
 
 
+def receive_receipt(receipt):
+    logger.info('Recieved receipt: %s', receipt)
+    if not validate_receipt_signature(receipt):
+        logger.info('Invalid receipt signature')
+        return False
+
+    # TODO - Add the receipt to an existing block in Temp folder
```

Comment:
@swapnilpawar31415 is this logic for temp receipt storage right?

Reply:
Yes.